src/ex_getln.c

```diff
@@ -4230,6 +4234,24 @@ f_getcmdline(typval_T *argvars UNUSED, typval_T *rettv)
     rettv->vval.v_string = get_cmdline_str();
 }
 
+/*
+ * Get current command line prompt.
+ */
+    static char_u *
+get_prompt(void)
+{
+    return current_prompt;
+}
+
+/*
+ * Set current command line prompt.
+ */
+    void
+set_prompt(char_u* str)
+{
+    vim_strncpy(current_prompt, str, sizeof(current_prompt) - 1);
```

Comment:
I think you need to free current_prompt when it is non-null before setting it.

Reply:
```c
static char_u	current_prompt[CMDBUFFSIZE + 1] = "";
```

`current_prompt` is static array.  Why free is needed?

Reply:
ah, I thought this allocates memory, but it uses strcpy, so I guess this should be fine.